SerialPrograms/Source/PokemonLGPE/Programs/ShinyHunting/PokemonLGPE_LegendaryReset.cpp

```diff
@@ -0,0 +1,188 @@
+#include "PokemonLGPE/Inference/Battles/PokemonLGPE_BattleArrowDetector.h"
+#include "PokemonLGPE/Inference/Sounds/PokemonLGPE_ShinySoundDetector.h"
+#include "PokemonLGPE/Programs/PokemonLGPE_GameEntry.h"
+#include "PokemonLGPE_LegendaryReset.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonLGPE{
+
+LegendaryReset_Descriptor::LegendaryReset_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonLGPE:LegendaryReset",
+        Pokemon::STRING_POKEMON + " LGPE", "Legendary Reset",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonLGPE/LegendaryReset.md",
+        "Shiny hunt Legendary Pokemon by resetting the game.",
+        FeedbackType::VIDEO_AUDIO,
+        AllowCommandsWhenRunning::DISABLE_COMMANDS,
+        {ControllerFeature::NintendoSwitch_RightJoycon},
+        FasterIfTickPrecise::NOT_FASTER
+    )
+{}
+
+struct LegendaryReset_Descriptor::Stats : public StatsTracker{
+    Stats()
+        : resets(m_stats["Resets"])
+        , shinies(m_stats["Shinies"])
+    {
+        m_display_order.emplace_back("Resets");
+        m_display_order.emplace_back("Shinies");
+    }
+    std::atomic<uint64_t>& resets;
+    std::atomic<uint64_t>& shinies;
+};
+std::unique_ptr<StatsTracker> LegendaryReset_Descriptor::make_stats() const{
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+LegendaryReset::LegendaryReset()
+    : TARGET(
+        "<b>Target:</b><br>",
+        {
+            {Target::mewtwo, "mewtwo", "Mewtwo, Articuno, Zapdos, Moltres"},
+            {Target::snorlax, "snorlax", "Snorlax"},
+            //{Target::snorlax2, "snorlax2", "Snorlax (w/Fuji)"},
+        },
+        LockMode::LOCK_WHILE_RUNNING,
+        Target::mewtwo
+    )
+    , GO_HOME_WHEN_DONE(true)
+    , NOTIFICATION_SHINY(
+        "Shiny Found",
+        true, true, ImageAttachmentMode::JPG,
+        {"Notifs", "Showcase"}
+    )
+    , NOTIFICATION_STATUS_UPDATE("Status Update", true, false, std::chrono::seconds(3600))
+    , NOTIFICATIONS({
+        &NOTIFICATION_SHINY,
+        &NOTIFICATION_STATUS_UPDATE,
+        &NOTIFICATION_PROGRAM_FINISH,
+    })
+{
+    PA_ADD_OPTION(TARGET);
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+bool LegendaryReset::run_battle(SingleSwitchProgramEnvironment& env, JoyconContext& context){
+    float shiny_coefficient = 1.0;
+    ShinySoundDetector shiny_detector(env.logger(), [&](float error_coefficient) -> bool{
+        shiny_coefficient = error_coefficient;
+        return true;
+    });
+    BattleArrowWatcher battle_started(COLOR_YELLOW);
+
+    env.log("Starting battle.");
+    switch (TARGET) {
+    case Target::mewtwo:
+        pbf_mash_button(context, BUTTON_A, 5000ms); //untested
+        break;
+    case Target::snorlax:
+        pbf_mash_button(context, BUTTON_A, 15000ms);
+        break;
+    case Target::snorlax2:
+        pbf_mash_button(context, BUTTON_A, 25000ms); //can't test, not really worth it
```

Comment:
Do these need to mash A for fixed durations? Can you mash A for a short amount of time, then mash B until you see the battle menu?